document-store/src/main/java/org/hypertrace/core/documentstore/GroupBy.java

```diff
@@ -0,0 +1,35 @@
+package org.hypertrace.core.documentstore;
+
+import java.util.List;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import org.apache.commons.lang3.StringUtils;
+import org.hypertrace.core.documentstore.expression.Expression;
+
+@Data
+@AllArgsConstructor
+public class GroupBy {
+  // TODO: Implement a builder for easily constructing the "group by" clause
+  private Expression key; // The key field whose distinct values would be returned
```

Comment:
The use of Expression seems to be very specific to GroupBy. 
Based on how its integrated, we can't do something like a select round(key) from collection.

So, should we rather enhance the Query class to allow for expressions in selections, orderby, filters too? This effectively means we cant reuse the current Query class in a backward compatible way. So may be a new method/api altogether.
cc: @aaron-steinfeld

Reply:
Yeah. For the first iteration, I only implemented expressions under GroupBy. I totally agree with you that we need to make it more generic (though it might break the existing compatibility).

Reply:
This class seems to encompass both selections and group bys, which is difficult to follow. I'd suggest starting with an assumed new API, let's see where it goes and then we can figure out if there's any reasonable way to make it compatible.

Reply:
Also, we should support (or leave room in the API to support), multiple levels of group by.

Reply:
What exactly do you mean by "multiple levels" of group by?
Did you mean, "GROUP BY colA, colB, colC"?

Reply:
yep, exactly

Reply:
Got it! Please refer to https://github.com/hypertrace/document-store/pull/58 for the API changes. Multiple grouping levels is supported in these API changes. However, need to figure out how to implement this for MongoDB.